signal_id/mask.py

```diff
@@ -413,19 +413,15 @@ def remove_small_regions(self, area_threshold=None, beam=None,
             # Eventually the unit should be included in get_pixel_scales
             # but this requires additional workarounds when used for Beam
             # objects
-            pixscale = get_pixel_scales(self._wcs) * u.deg
+            pixscale = get_pixel_scales(self._wcs)
+
             # Now get the pixel beam area
-            area_threshold = (beam.sr.to(u.deg**2) / pixscale**2.).value
+            area_threshold = (beam.as_tophat_kernel(pixscale).array > 0).sum()
```

Comment:
`np.count_nonzero` is slightly more efficient than `.sum()` for boolean array summation like this.  This hint courtesy Julian Taylor